### PR TITLE
CDK: Normalize environment names

### PR DESCRIPTION
**Pull request by kylelaker-ccpo**
This applies the changes in dod-ccpo/atat-web-api#193 to be consumed by
the front-end stack. dod-ccpo/atat-web-api#193 should be merged first
though that isn't necessarily a requirement.

### CODE REVIEW

File: cdk/util.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+export function lowerCaseEnvironmentId(id: string): string {
+  return id.toLowerCase().replace(normalizationRejectionRegex, "");
+}
+
+/**
+ * Helper type guard to check whether an input is a string
+ * @param str the input to test
+ * @returns true if the input is a non-empty string and false otherwise
+ */
+export function isString(str: unknown): str is string {
```
Comment (tonysumner-ccpo) [inline cdk/util.ts:46]: If I not mistaken this could be rewritten as `return str && typeof str === "string"`

File: cdk/index.ts [hunk 2]
```diff
@@ -48,8 +48,15 @@ class StaticSiteStack extends cdk.Stack {
 }
 
 const app = new cdk.App();
-const environmentId = app.node.tryGetContext("EnvironmentId");
-const spaStack = new StaticSiteStack(app, environmentId + "StaticSpa", {
+const environmentParam = app.node.tryGetContext("EnvironmentId");
```
Comment (zachclark-ccpo) [inline cdk/index.ts:51]: Should we change this to:
```
const environmentParam = app.node.tryGetContext("EnvironmentId") ?? app.node.tryGetContext("TicketId");
```
To be consistent with atat-web-api?

**Review by zachclark-ccpo — APPROVED** [review]

File: cdk/util.ts (hunk 1 above)
Comment (kylelaker-ccpo) [reply cdk/util.ts:46]: Good catch!

File: cdk/index.ts (hunk 2 above)
Comment (kylelaker-ccpo) [reply cdk/index.ts:51]: No. ``TicketId` was never a thing on this repo so there's no need to add that backwards compatibility. We'll let folks be broken for a little while on `atat-web-api` but there's no reason to add legacy code if that legacy never existed.

**Pull request merged**